prometheus/registry.go

```diff
@@ -11,232 +11,253 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// Copyright (c) 2013, The Prometheus Authors
-// All rights reserved.
-//
-// Use of this source code is governed by a BSD-style license that can be found
-// in the LICENSE file.
-
 package prometheus
 
 import (
-	"bytes"
-	"compress/gzip"
 	"errors"
 	"fmt"
-	"io"
-	"net/http"
-	"net/url"
 	"os"
 	"sort"
-	"strings"
 	"sync"
 
 	"github.com/golang/protobuf/proto"
-	"github.com/prometheus/common/expfmt"
+	"github.com/hashicorp/go-multierror"
```

Comment:
Since we have a number of dependencies anyway, I see the cost of adding two more as the lesser evil compared to implementing the 47th variant of a multi error. (Plus, this is one that could be vendored internally, once we go for it.)

But I'm just repeating myself, and so are you.

Do others have additional arguments here or an opinion?